domain/application/state/state.go

```diff
@@ -1306,16 +1307,31 @@ WHERE name = $applicationDetails.name
 	return nil
 }
 
+// checkApplicationAlive checks if the application exists and is alive.
+//   - If the application is not alive, [applicationerrors.ApplicationNotAlive] is returned.
+//   - If the application is not found, [applicationerrors.ApplicationNotFound]
+//     is returned.
+func (st *State) checkApplicationAlive(ctx context.Context, tx *sqlair.TX, appUUID coreapplication.ID) error {
+	return st.checkApplicationLife(ctx, tx, appUUID, domainlife.Alive)
+}
+
 // checkApplicationNotDead checks if the application exists and is not dead. It's
 // possible to access alive and dying applications, but not dead ones.
 //   - If the application is dead, [applicationerrors.ApplicationIsDead] is returned.
 //   - If the application is not found, [applicationerrors.ApplicationNotFound]
 //     is returned.
-func (st *State) checkApplicationNotDead(ctx context.Context, tx *sqlair.TX, ident applicationID) error {
+func (st *State) checkApplicationNotDead(ctx context.Context, tx *sqlair.TX, appUUID coreapplication.ID) error {
+	return st.checkApplicationLife(ctx, tx, appUUID, domainlife.Dying)
+}
+
+// checkApplicationLife checks if the application exists and its life has not
+// advanced beyond the specified allowed life.
+func (st *State) checkApplicationLife(ctx context.Context, tx *sqlair.TX, appUUID coreapplication.ID, allowed domainlife.Life) error {
```

Comment:
We should advise not to use this, instead use the higher level method i.e. `checkApplicationNotDead`